### ISSUE TITLE
Empty typing.Tuple

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/cpython#91137]
BPO | [46981](https://bugs.python.org/issue46981)
--- | :---
Nosy | @gvanrossum, @serhiy-storchaka, @JelleZijlstra, @Fidget-Spinner
PRs | <li>python/cpython#31836</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2022-03-17.08:20:59.454>
created_at = <Date 2022-03-11.08:21:12.398>
labels = ['library', '3.11']
title = 'Empty typing.Tuple'
updated_at = <Date 2022-04-08.05:07:15.239>
user = 'https://github.com/serhiy-storchaka'
```

bugs.python.org fields:
```python
activity = <Date 2022-04-08.05:07:15.239>
actor = 'JelleZijlstra'
assignee = 'none'
closed = True
closed_date = <Date 2022-03-17.08:20:59.454>
closer = 'serhiy.storchaka'
components = ['Library (Lib)']
creation = <Date 2022-03-11.08:21:12.398>
creator = 'serhiy.storchaka'
dependencies = []
files = []
hgrepos = []
issue_num = 46981
keywords = ['patch']
message_count = 5.0
messages = ['414892', '414947', '414986', '415391', '416958']
nosy_count = 4.0
nosy_names = ['gvanrossum', 'serhiy.storchaka', 'JelleZijlstra', 'kj']
pr_nums = ['31836']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue46981'
versions = ['Python 3.11']
```

</p></details>

**Comment by serhiy-storchaka** [comment]
We get a report about broken pickling of empty `typing.Tuple` #94245. Shallow and deep copying are broken too.
```
>>> copy.copy(Tuple[()])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/serhiy/py/cpython3.10/Lib/copy.py", line 102, in copy
    return _reconstruct(x, None, *rv)
  File "/home/serhiy/py/cpython3.10/Lib/copy.py", line 265, in _reconstruct
    y = func(*args)
  File "/home/serhiy/py/cpython3.10/Lib/typing.py", line 312, in inner
    return func(*args, **kwds)
  File "/home/serhiy/py/cpython3.10/Lib/typing.py", line 1233, in __getitem__
    params = tuple(_type_check(p, msg) for p in params)
  File "/home/serhiy/py/cpython3.10/Lib/typing.py", line 1233, in <genexpr>
    params = tuple(_type_check(p, msg) for p in params)
  File "/home/serhiy/py/cpython3.10/Lib/typing.py", line 176, in _type_check
    raise TypeError(f"{msg} Got {arg!r:.100}.")
TypeError: Tuple[t0, t1, ...]: each t must be a type. Got ().
>>> copy.deepcopy(Tuple[()])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/serhiy/py/cpython3.10/Lib/copy.py", line 172, in deepcopy
    y = _reconstruct(x, memo, *rv)
  File "/home/serhiy/py/cpython3.10/Lib/copy.py", line 265, in _reconstruct
    y = func(*args)
  File "/home/serhiy/py/cpython3.10/Lib/typing.py", line 312, in inner
    return func(*args, **kwds)
  File "/home/serhiy/py/cpython3.10/Lib/typing.py", line 1233, in __getitem__
    params = tuple(_type_check(p, msg) for p in params)
  File "/home/serhiy/py/cpython3.10/Lib/typing.py", line 1233, in <genexpr>
    params = tuple(_type_check(p, msg) for p in params)
  File "/home/serhiy/py/cpython3.10/Lib/typing.py", line 176, in _type_check
    raise TypeError(f"{msg} Got {arg!r:.100}.")
TypeError: Tuple[t0, t1, ...]: each t must be a type. Got ().
```
All this can be fixed by backporting this change. But it will break pyanalyze.

**Comment by amitbad** [comment]
I was using python 3.10 and getting the same error while using Gemini-Pro-Vision. 

This resolved my issue - pip install pydantic==1.10.13
